app/questionnaire/path_finder.py

```diff
@@ -180,6 +180,10 @@ def get_blocks(self):
             if repeating_rule:
                 no_of_repeats = evaluate_repeat(repeating_rule, self.answer_store)
 
+            if no_of_repeats > 50:
+                logger.error('Excessive number of repeats found [%s] capping at 50', no_of_repeats)
+                no_of_repeats = 50
```

Comment:
As previously mentioned this is just enforcing a system limit to prevent crashing. The validation that you are talking about is a survey specific validation.